### PR TITLE
Feature 342 ask refund

### LINKED ISSUE #342: As a funder I want to be able to ask for a refund
### Expected behaviour
When an issue is funded and for some reason is not getting resolved, the funders funds would be "lost". A funder should have the opportunity to request a refund. In the first version a manual approval is required to refund the funder. 

The initial goal of this request is to avoid locked funding.

### LINKED ISSUE #343: As an admin I would like to be able to get an overview of all the refund requests and be able to approve or decline each request
As an admin I would like to be able to get an overview of all the refund requests and be able to approve or decline each request.
If a request is approve , the GitHub comments made by the fundrequest-notifier should be updated to indicate to the developers that a refund has been approved and that this issue is no longer funded.

related issue : https://github.com/FundRequest/platform/issues/342

### LINKED ISSUE #494: When a request has no funds attached (due to a refund) the request should not be visible in the overal overview
When a request has no funds attached (due to a refund) the request should not be visible in the overal overview.

But it should remain in personal overview.

### PR DESCRIPTION
**Pull request by nico-ptrs**
Fixes #342
Fixes #343 
Fixes #494

### CODE REVIEW

File: tweb/src/main/java/io/fundreqest/platform/tweb/fund/FundRestController.java [hunk 1]
```diff
@@ -18,7 +18,7 @@ public FundRestController(final FundRequestContractsService fundRequestContracts
         this.fundRequestContractsService = fundRequestContractsService;
     }
 
-    @RequestMapping("/allowed-tokens")
+    @RequestMapping("/fund/allowed-tokens")
```
Comment (davyvanroy) [inline tweb/src/main/java/io/fundreqest/platform/tweb/fund/FundRestController.java]: frontend changed as well?

File: tweb/src/main/java/io/fundreqest/platform/tweb/infrastructure/mav/Initializer.java [hunk 2]
```diff
@@ -10,32 +11,39 @@
 @Component
 public class Initializer {
 
+    private final boolean initializeCacheOnStartup;
+
+    public Initializer(@Value("${io.fundrequest.cache.initialize-on-startup:true}") final boolean initializeCacheOnStartup) {
```
Comment (davyvanroy) [inline tweb/src/main/java/io/fundreqest/platform/tweb/infrastructure/mav/Initializer.java]: should be default false and property added to the servers

File: tweb/src/main/java/io/fundreqest/platform/tweb/fund/FundRestController.java (hunk 1 above)
Comment (nico-ptrs) [reply tweb/src/main/java/io/fundreqest/platform/tweb/fund/FundRestController.java]: This is actually a leftover from a rest service I created and removed during development.
The 'fund/' is taken out from the requestMapping at class level and put at method level. So nothing changes for the frontend.

File: admin-web/src/main/resources/application-local.properties [hunk 3]
```diff
@@ -42,5 +42,5 @@ io.fundrequest.contract.fund-request.address=0xC16a102813B7bD98b0BEF2dF28FFCaf1F
 io.fundrequest.contract.tokenAddress-whitelist-precondition.address=0x401EccAbA12E6466a1c838917DBe2D2669D64E34
 io.fundrequest.contract.tokenAddress.address=0x02F96eF85cAd6639500CA1cc8356F0b5CA5bF1D2
 
-io.fundrequest.badge.basepath=http://localhost:8080
+io.fundrequest.platform.base-path=http://localhost:8080
```
[the remark this answers is not in the document]
Comment (nico-ptrs) [reply admin-web/src/main/resources/application-local.properties]: - [x] Checked

File: core/src/main/java/io/fundrequest/core/request/fund/domain/RefundRequest.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+@Entity
+@Table(name = "refund_request")
+@Getter
+@Setter
+@ToString
+@EqualsAndHashCode(exclude = "id", callSuper = false)
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
```
[the remark this answers is not in the document]
Comment (nico-ptrs) [reply core/src/main/java/io/fundrequest/core/request/fund/domain/RefundRequest.java]: Lombok for the win!

File: tweb/src/main/java/io/fundreqest/platform/tweb/infrastructure/mav/Initializer.java (hunk 2 above)
Comment (nico-ptrs) [reply tweb/src/main/java/io/fundreqest/platform/tweb/infrastructure/mav/Initializer.java]: - [x] Done in code
- [x]  Done on servers

File: admin-web/src/main/java/io/fundrequest/platform/admin/refund/RefundController.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+                refundModerationService.decline(refundRequestId);
+                break;
+            default:
+                throw new BadRequestException("Action not allowed");
+        }
+
+        return new ModelAndView(new RedirectView("/refunds", true));
+    }
+
+    @ExceptionHandler(BadRequestException.class)
```
[the remark this answers is not in the document]
Comment (nico-ptrs) [reply admin-web/src/main/java/io/fundrequest/platform/admin/refund/RefundController.java]: - [x] done

File: db/src/main/resources/db/migration/V20180614125037__create_refund_table.sql [hunk 6]
```diff
@@ -0,0 +1,16 @@
+CREATE TABLE refund
```
[the remark this answers is not in the document]
Comment (nico-ptrs) [reply db/src/main/resources/db/migration/V20180614125037__create_refund_table.sql]: - [x] Done

**Comment by nico-ptrs** [comment]
@davyvanroy Re-review please

**Pull request merged**